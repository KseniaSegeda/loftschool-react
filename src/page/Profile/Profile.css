.profile {
    background-image: url("../../assert/login-background.jpg");
    background-size: cover;
    height: calc(100vh - 65px);
}

.profile__container {
    display: flex;
    justify-content: center;
}

.profile__window {
    padding: 44px 60px;
    margin-top: 48px;
    margin-bottom: 48px;
}

.profile-title {
    text-align: center;
    margin-bottom: 40px;
}

.profile-form__block {
    position: relative;
    padding: 16px 32px;
    margin-bottom: 20px;
}

.form__field {
   margin-bottom: 15px;
}
.form__field #cardName{
    margin-bottom: 0;
}

