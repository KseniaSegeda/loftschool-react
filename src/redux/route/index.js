export * from "./actions";
export * from "./saga";
export * from "./select";
export { default } from './reducer';
