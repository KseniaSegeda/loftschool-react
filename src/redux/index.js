import { combineReducers } from 'redux';
import auth from './auth/reducer';
import pay from './payments/reducer';
import route from './route/reducer'

export default combineReducers({auth, pay, route})
