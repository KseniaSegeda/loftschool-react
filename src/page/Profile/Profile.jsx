import React, {useEffect, useState} from "react";
import {Button, Grid, Paper, TextField, Typography} from "@material-ui/core";
import {MCIcon} from 'loft-taxi-mui-theme';
import './Profile.css';
import DateFnsUtils from '@date-io/date-fns';
import {DatePicker, MuiPickersUtilsProvider,} from '@material-ui/pickers';
import {pendingGetCard, pendingSetCard, getCard} from "../../redux/payments/";
import {connect} from "react-redux";


const Profile = (props) => {
    const {token, pendingSetCard, pendingGetCard, cardNumber, expiryDate, cardName, cvc} = props;
    const [isUpdateCard, setIsUpdateCard] = useState(true);
    const [formFields, setFormField] = useState({
        cardNumber: "",
        expiryDate: +new Date(),
        cardName: "",
        cvc: "",
        token: token
    });

    useEffect(() => {
        if (isUpdateCard) {
            pendingGetCard();
            setIsUpdateCard(false)
        }
        setFormField({cardNumber, expiryDate, cardName, cvc})
    }, [isUpdateCard, cardNumber, expiryDate, cardName, cvc, pendingGetCard]);

    const onChange = (e) => {
        setFormField({
            ...formFields,
            [e.target.name]: e.target.value,
        });
    };

    const onDateChange = date => {
        setFormField({
            ...formFields,
            expiryDate: date
        })
    }


    const submit = (e) => {
        e.preventDefault();
        pendingSetCard(formFields);
    };
    return <div className="profile" data-testid="profile">
        <div className="profile__container">
            <Paper className="profile__window">
                <div className="profile-title">
                    <div className="profile-title__header">
                        <Typography variant="h4">Профиль</Typography>
                    </div>
                    <div className="profile-title__subtext">
                        <Typography variant="body1">Способ оплаты</Typography>
                    </div>
                </div>
                <form onSubmit={submit}>
                    <Grid container>
                        <Grid item xs={12}>
                            <Grid container spacing={4} justify="center">
                                {props.isLoading ?
                                     <>
                                        <Grid item xs={6}>
                                            <Paper elevation={4} className="profile-form__block">
                                                <div className="icon">
                                                    <MCIcon/>
                                                </div>
                                                <div className="form__field">
                                                    <TextField
                                                        id="cardNumber"
                                                        label="Номер карты"
                                                        type="cardNumber"
                                                        fullWidth
                                                        placeholder="0000 0000 0000 0000"
                                                        name="cardNumber"
                                                        value={formFields.cardNumber}
                                                        onChange={onChange}
                                                        required
                                                    />
                                                </div>
                                                <div className="form__field">
                                                    <MuiPickersUtilsProvider utils={DateFnsUtils}>
                                                        <DatePicker
                                                            name="expiryDate"
                                                            id="expiryDate"
                                                            value={formFields.expiryDate}
                                                            onChange={onDateChange}
                                                            openTo="year"
                                                            views={["year", "month"]}
                                                            format="MM/yy"
                                                            InputLabelProps={{shrink: true}}
                                                            autoOk={true}
                                                            fullWidth
                                                            required
                                                        />
                                                    </MuiPickersUtilsProvider>
                                                </div>
                                            </Paper>
                                        </Grid>
                                        <Grid item xs={6}>
                                            <Paper elevation={4} className="profile-form__block">
                                                <div className="form__field">
                                                    <TextField
                                                        id="cardName"
                                                        label="Имя владельца"
                                                        type="cardName"
                                                        fullWidth
                                                        name="cardName"
                                                        placeholder="USER NAME"
                                                        value={formFields.cardName}
                                                        onChange={onChange}
                                                        required
                                                    />
                                                </div>
                                                <div className="form__field">
                                                    <TextField
                                                        id="cvc"
                                                        label="CVC"
                                                        type="cvc"
                                                        fullWidth
                                                        name="cvc"
                                                        value={formFields.cvc}
                                                        onChange={onChange}
                                                        required
                                                    />
                                                </div>
                                            </Paper>
                                        </Grid>
                                     </> : <Grid item xs={6}>loading....</Grid>
                                }
                                <div>
                                    <Button type="submit" variant="contained" color="primary">
                                        Сохранить
                                    </Button>
                                </div>

                            </Grid>
                        </Grid>
                    </Grid>
                </form>
            </Paper>
        </div>
    </div>
}

export default connect(
    getCard,
    {pendingSetCard, pendingGetCard}
)(Profile);
