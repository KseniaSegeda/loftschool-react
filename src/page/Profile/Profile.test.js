import {createStore} from "redux";
import rootReducer from "../../redux";
import ReactDOM from "react-dom";
import React from "react";
import Profile from "./Profile";
import renderer from "react-test-renderer";
import {render} from "@testing-library/react";

// const FAKE_DATA_CARD = {
//     cardNumber: '1111 1111 1111 1111',
//     expiryDate: '10/24',
//     cardName: 'TEST TEST',
//     cvc: '111'
// }

describe('Profile', () => {
    let store;
    beforeAll(() => {
        store = createStore(rootReducer);
    })

    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(global.wrapperProvider(<Profile/>, store), div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('render correctly', function () {
        const tree = renderer.create(global.wrapperProvider(<Profile />, store)).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('card not set -> loading', function () {
        const { getByText, getByTestId } = render(
            global.wrapperProvider(<Profile />, store)
        );
        expect(getByText('loading....')).toBeInTheDocument();
    });
});
