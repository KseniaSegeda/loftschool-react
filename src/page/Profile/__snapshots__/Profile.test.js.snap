// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile render correctly 1`] = `
<div
  className="profile"
  data-testid="profile"
>
  <div
    className="profile__container"
  >
    <div
      className="MuiPaper-root profile__window MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="profile-title"
      >
        <div
          className="profile-title__header"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4"
          >
            Профиль
          </h4>
        </div>
        <div
          className="profile-title__subtext"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Способ оплаты
          </p>
        </div>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-justify-xs-center"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              >
                loading....
              </div>
              <div>
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                  disabled={false}
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="submit"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Сохранить
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
